ocaml/testsuite/tests/typing-modes/class.ml

```diff
@@ -0,0 +1,149 @@
+(* TEST
```

Comment:
I was expecting to see some tests around the mutability mode of instance variables. It's definitely always going to be the legacy mode. But we still need to be careful: we must check that anything written into an instance variable is uncontended, because -- unusually -- we can write to a *contended* instance variable. So I think a corner case exists here that doesn't exist elsewhere. Getting it wrong would allow us to smuggle a contended value into an uncontended one by storing it in an instance variable -- bad.

Reply:
Similar to above: the type checker doesn't even talk about the `mutable(m0)` of instance variables.

Reply:
Similar to above, the type checker doesn't talk about the modes of instance variables.

But you are right that we should test that it's disallowed to write non-legacy value into a instance variable.

Reply:
Yes, I think that would capture the kind of thing I'm worried about here. We need to make sure everything that comes out of a class is legacy, *and* we need to make sure that everything that goes in is legacy, too.